# Query centroids of a place (town/region/...)

In [1]:
from geopy.geocoders import Nominatim
geolocator = Nominatim(user_agent="Test")

## Getting the centroid coordinates of a place

When making a search with *Nominatim*, the result already includes the longitude and latitude :

In [2]:
location = geolocator.geocode("Bayern") # Bade Wurtemberg
print(location.address)
print(location.latitude, location.longitude)

Bayern, Deutschland
48.9467562 11.4038717


## Getting a precise address from the coordinates

If GPS coordinates are not sufficient, we might need to transform them to a precise street number, we just have to reverse-geocoding on the acquired GPS coordinates :

In [3]:
location = geolocator.reverse((location.latitude, location.longitude))
print(location)

Kipfenberg, Landkreis Eichstätt, Obb, Bayern, 85110, Deutschland


## Testing the given value

### Baden-Württemberg
I entered the given coordinates for the centroid of Baden-Württemberg (**48.6296972**, **9.1949534**) in Google Maps and made an itinary between this centroid and the one given by Google Maps :

![Distance between osm and google maps centroids for Baden-Württemberg](./img/google_maps_distance_osm_gm_bw_centroids.png)

We can see that there is only **14.4km** between the two points. If we consider *Baden-Württemberg* is roughly **200km** wide, we have a difference of **7.2%**.


## Bayern

Using OSM coordinates of (**48.9467562**, **11.4038717**) :

![Distance between osm and google maps centroids for Bayern](./img/google_maps_distance_osm_gm_bayern_centroids.png)

We have **22.4km** between the two centroids. Considering *Bayern* is roughly **300km** long, we have a difference of **7.5%**.

## Conclusion on precision of centroids

The values used here are obviously not very accurate and only two examples are usedbut we can still see that, each time, the given centroid is localted roughly in the center of the requested region.

The difference with Google Maps can maybe be explained by a difference of precision in the borders.